src/main/scala/mesosphere/marathon/state/LinuxInfo.scala

```diff
@@ -13,30 +14,106 @@ import com.wix.accord._
 case class Seccomp(profileName: Option[String], unconfined: Boolean)
 
 /**
-  * Defines the linux runtime the container will run under.   This could include things such as seccomp, linux capabilities,
+  * Defines the IPC configuration for the instance or executor this is associated with.
+  *
+  * @param ipcMode The IPC mode configured
+  * @param shmSize The shared memory size in MB when using private IPC mode
+  */
+case class IPCInfo(ipcMode: state.IpcMode, shmSize: Option[Int])
+
+/**
+  * Defines the linux runtime the container will run under. This could include things such as seccomp, linux capabilities,
   * and shared pid namespaces.
   *
   * @param seccomp The seccomp mode to use, either unconfined or which profile.
+  * @param ipcInfo The IPCConfig to use
   */
-case class LinuxInfo(seccomp: Option[Seccomp])
+case class LinuxInfo(seccomp: Option[Seccomp], ipcInfo: Option[IPCInfo])
 
 object LinuxInfo {
 
-  /*
-  rules:  if seccomp not defined = valid
-          if profile is empty unconfined must be true to be valid
-          if profile is not empty unconfined must be false to be valid
-   */
-  val validLinuxInfo =
-    new Validator[LinuxInfo] {
-      override def apply(linuxInfo: LinuxInfo): Result = {
-        linuxInfo.seccomp match {
-          case Some(seccomp) =>
-            if (seccomp.profileName.isDefined && seccomp.unconfined) Failure(Set(RuleViolation(linuxInfo, "Seccomp unconfined can NOT be true when Profile is defined")))
-            else if (seccomp.profileName.isEmpty && !seccomp.unconfined) Failure(Set(RuleViolation(linuxInfo, "Seccomp unconfined must be true when Profile is NOT defined")))
-            else Success
-          case None => Success
-        }
+  //  /*
```

Comment:
This is an interesting combination of comment chars. Can you use either `/*` or `//`?

Reply:
I blame IntelliJ ;)